src/napari/components/overlays/base.py

```diff
@@ -23,6 +25,8 @@ class Overlay(EventedModel):
         The rendering order of the overlay: lower numbers get rendered first.
     """
 
+    model_config = ConfigDict(validate_assignment=True)
+
```

Comment:
You can (maybe should?) do:

```suggestion
    model_config = EventedModel.mode_config | ConfigDict(validate_assignment=True)
```

OR as I understand it it will override defaults? This might be more relevant for when we want to have the same eventedmodel setup for all napari, which requires some extra configs (like for example `validate_defaults=True` and the json schema for np.ndarray).